xarray/tests/test_computation.py

```diff
@@ -789,6 +789,70 @@ def func(x):
     assert_identical(expected, actual)
 
 
+da = xr.DataArray(
+    np.random.random((3, 21, 4)),
+    coords={"time": pd.date_range("2000-01-01", freq="1D", periods=21)},
+    dims=("a", "time", "x"),
+)
+
+arrays = [
+    da.isel(time=range(0, 18)),
+    da.isel(time=range(2, 20)).rolling(time=3, center=True).mean(dim="time"),
+    xr.DataArray([0, np.nan, 1, 2, np.nan, 3, 4, 5, np.nan, 6, 7], dims="time"),
+    xr.DataArray([1, 1, np.nan, 2, np.nan, 3, 5, 4, 6, np.nan, 7], dims="time"),
+    xr.DataArray([[1, 2], [1, np.nan]], dims=["x", "time"]),
+    xr.DataArray([[1, 2], [np.nan, np.nan]], dims=["x", "time"]),
+]
+
+array_tuples = [
+    (arrays[0], arrays[0]),
+    (arrays[0], arrays[1]),
+    (arrays[1], arrays[1]),
+    (arrays[2], arrays[2]),
+    (arrays[2], arrays[3]),
+    (arrays[3], arrays[3]),
+    (arrays[4], arrays[4]),
+    (arrays[4], arrays[5]),
+    (arrays[5], arrays[5]),
+]
+
+
+@pytest.mark.parametrize("da_a, da_b", array_tuples)
+@pytest.mark.parametrize("dim", [None, "time"])
+def test_cov(da_a, da_b, dim):
+    def pandas_cov(ts1, ts2):
+        """Ensure the ts are aligned and missing values ignored"""
+        ts1, ts2 = xr.align(ts1, ts2)
+        valid_values = ts1.notnull() & ts2.notnull()
+
+        ts1 = ts1.where(valid_values, drop=True)
+        ts2 = ts2.where(valid_values, drop=True)
```

Comment:
I would have thought that we should be testing that we get a similar result from pandas' `cov` and xarray `cov` _without_ much preparation. Can you help me understand why we need to drop values here? What's the test case that fails when we don't have these lines?